bus/bus.go

```diff
@@ -379,54 +379,58 @@ func (b *Bus) contractsIDHandlerDELETE(jc jape.Context) {
 	jc.Check("couldn't remove contract", b.cs.RemoveContract(id))
 }
 
-func (b *Bus) hostsetsHandler(jc jape.Context) {
-	hostSets, err := b.hss.HostSets()
-	if jc.Check("couldn't load host sets", err) != nil {
+func (b *Bus) contractSetHandler(jc jape.Context) {
+	contractSets, err := b.css.ContractSets()
+	if jc.Check("couldn't load contract sets", err) != nil {
 		return
 	}
-	jc.Encode(hostSets)
+	jc.Encode(contractSets)
 }
 
-func (b *Bus) hostsetsNameHandlerGET(jc jape.Context) {
-	hostSet, err := b.hss.HostSet(jc.PathParam("name"))
+func (b *Bus) contractSetsNameHandlerGET(jc jape.Context) {
+	hostSet, err := b.css.ContractSet(jc.PathParam("name"))
 	if jc.Check("couldn't load host set", err) != nil {
 		return
 	}
 	jc.Encode(hostSet)
 }
 
-func (b *Bus) hostsetsNameHandlerPUT(jc jape.Context) {
-	var hosts []consensus.PublicKey
-	if jc.Decode(&hosts) != nil {
+func (b *Bus) contractSetsNameHandlerPUT(jc jape.Context) {
+	var contracts []types.FileContractID
+	if jc.Decode(&contracts) != nil {
 		return
 	}
-	jc.Check("couldn't store host set", b.hss.SetHostSet(jc.PathParam("name"), hosts))
+	jc.Check("couldn't store host set", b.css.SetContractSet(jc.PathParam("name"), contracts))
 }
 
-func (b *Bus) hostsetsContractsHandler(jc jape.Context) {
-	hosts, err := b.hss.HostSet(jc.PathParam("name"))
+func (b *Bus) contractSetContractsHandler(jc jape.Context) {
+	setContracts, err := b.css.ContractSet(jc.PathParam("name"))
 	if jc.Check("couldn't load host set", err) != nil {
 		return
 	}
-	latest := make(map[consensus.PublicKey]rhpv2.Contract)
 	all, err := b.cs.Contracts()
 	if jc.Check("couldn't load contracts", err) != nil {
 		return
 	}
+	allMap := make(map[types.FileContractID]*rhpv2.Contract)
 	for _, c := range all {
-		if old, ok := latest[c.HostKey()]; !ok || c.EndHeight() > old.EndHeight() {
-			latest[c.HostKey()] = c
-		}
-	}
-	contracts := make([]Contract, len(hosts))
-	for i, host := range hosts {
-		contracts[i].HostKey = host
-		hi, err := b.hdb.Host(host)
-		if err == nil {
-			contracts[i].HostIP = hi.NetAddress()
-		}
-		if c, ok := latest[host]; ok {
-			contracts[i].ID = c.ID()
+		allMap[c.ID()] = &c
+	}
+	var contracts []Contract
+	for _, fcid := range setContracts {
+		c, exists := allMap[fcid]
+		if exists {
+			// TODO: The number of contract types we have slowly
+			// grows out of control. We should find a solution for
+			// this.
```

Comment:
😬 

"Contract" is definitely an overloaded term... `bus`, `worker`, `autopilot` all have their own definition, not to mention `rhp` (and `types`!)